SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityDeployChecker.java

```diff
@@ -274,39 +274,46 @@ private void finishDeploy(SingularityRequestWithState requestWithState, Optional
 
+      SingularityPendingRequestBuilder builder = new SingularityPendingRequestBuilder()
+          .setRequestId(requestId)
+          .setDeployId(deploy.get().getId())
+          .setTimestamp(deployResult.getTimestamp())
+          .setUser(pendingDeploy.getDeployMarker().getUser())
+          .setCmdLineArgsList(runNowRequest.getCommandLineArgs())
+          .setRunId(runNowRequest.getRunId().or(Optional.of(UUID.randomUUID().toString())))
+          .setSkipHealthchecks(runNowRequest.getSkipHealthchecks().or(request.getSkipHealthchecks()))
+          .setMessage(runNowRequest.getMessage()
+              .or(pendingDeploy.getDeployMarker().getMessage()))
+          .setResources(runNowRequest.getResources())
+          .setEnvOverrides(runNowRequest.getEnvOverrides())
+          .setRunAt(runNowRequest.getRunAt());
+
+      PendingType pendingType = null;
       if (request.isScheduled()) {
         if (activeTasks.isEmpty()) {
-          PendingType pendingType = canceledOr(deployResult.getDeployState(), PendingType.IMMEDIATE);
-          requestManager.addToPendingQueue(new SingularityPendingRequest(requestId, deployId, timestamp, user, pendingType, commandLineArgs, runId, skipHealthChecks, message, Optional.absent(), resources, runNowRequest.getRunAt()));
+          pendingType = PendingType.IMMEDIATE;
         } else {
           // Don't run scheduled task over a running task. Will be picked up on the next run.
-          PendingType pendingType = canceledOr(deployResult.getDeployState(), PendingType.NEW_DEPLOY);
-          requestManager.addToPendingQueue(new SingularityPendingRequest(requestId, deployId, timestamp, user, pendingType, commandLineArgs, runId, skipHealthChecks, message, Optional.absent(), resources, runNowRequest.getRunAt()));
+          pendingType = PendingType.NEW_DEPLOY;
         }
       } else if (!request.isLongRunning()) {
         if (request.getInstances().isPresent()
             && (activeTasks.size() + pendingTasks.size() < request.getInstances().get())) {
-          PendingType pendingType = canceledOr(deployResult.getDeployState(), PendingType.ONEOFF);
-          requestManager.addToPendingQueue(new SingularityPendingRequest(requestId, deployId, timestamp, user, pendingType, commandLineArgs, runId, skipHealthChecks, message, Optional.absent(), resources, runNowRequest.getRunAt()));
+          pendingType = PendingType.ONEOFF;
         } else {
           // Don't run one-off / on-demand task when already at instance count cap
-          PendingType pendingType = canceledOr(deployResult.getDeployState(), PendingType.NEW_DEPLOY);
-          requestManager.addToPendingQueue(new SingularityPendingRequest(requestId, deployId, timestamp, user, pendingType, commandLineArgs, runId, skipHealthChecks, message, Optional.absent(), resources, runNowRequest.getRunAt()));
+          pendingType = PendingType.NEW_DEPLOY;
         }
       }
+      if (pendingType != null) {
```

Comment:
Should we be handling the case where pendingType == null here?

Reply:
I don't think so. This is just a small refactor of the previous code, which would do nothing in the case where the new code has pendingType == null.